### DISCUSSION TITLE
are there any security implications using any of the Roslyn API ?

### DISCUSSION THREAD

**Discussion by birojnayak** [dotnet/roslyn#68595] [Q&A] [answered]
Hi Folks,
I wanted to understand if there are any security implications using Roslyn API ? I am using it to generate a compilation unit and generating syntax tree/ semantic models. Will it be possible if there are any powershell script or any script which can be executed as a part of any API call ? Or any method invocation from a binary could potentially happen ?

In short if I am using same sandbox environment for analyzing multiple source code , will it  pose any risk ? If yes, what are they ?

Thanks !

**Comment by CyrusNajmabadi** [comment]
Yes.  We run arbitrary plugins as part of our compilation system.  There are no restrictions on what those plugins can do.

> In short if I am using same sandbox environment for analyzing multiple source code , will it pose any risk ? If yes, what are they ?

It povides as much risk as your sandbox provides.   You can only trust roslyn as much as you trust your sandbox.

**Reply by birojnayak** [reply]
@CyrusNajmabadi thank you for responding ! 

Could you please link documentation or code to understand a bit around "arbitary plugins" ? (Would be very helpful). 
Secondly, can a power shell script/ or any script can be executed during source compilation to emit dll ?

**Reply by CyrusNajmabadi** [reply, marked as answer]
Look up analyzers and source generators. We load them and they can execute arbitrary code. We do no sandboxing ourselves. That's the long and short of it. If you want protection, run in an external sandbox :-)
